ejb3/src/main/java/org/jboss/as/ejb3/subsystem/EJB3SubsystemAdd.java

```diff
@@ -594,13 +597,24 @@ private static void addClusteringServices(final OperationContext context, final
             ServiceInstaller installer = new ServiceInstaller() {
                 @Override
                 public ServiceController<?> install(RequirementServiceTarget target) {
-                    ServiceBuilder<?> builder = targetFactory.get().createSingletonServiceTarget(target).addService();
-                    return builder.setInstance(org.jboss.msc.Service.newInstance(builder.provides(CLUSTERED_SINGLETON_CAPABILITY.getCapabilityServiceName()), Boolean.TRUE))
-                            .setInitialMode(ServiceController.Mode.ON_DEMAND)
-                            .install();
+                    // Install on-demand singleton service
+                    // We don't want this to start until a deployment requires it
```

Comment:
Note to reviewers/maintainers: Unlike traditional MSC APIs, the wildfly-service stuff creates ON_DEMAND services by default. That's why there's no code making these on-demand to satisfy what the comment says.